### PR TITLE
A list of videos supported by AMP

### PR DESCRIPTION
**Pull request by kul3r4**
This addresses [issue:](https://github.com/ampproject/amp-by-example/issues/41)

### CODE REVIEW

**Comment by kul3r4** [comment]
#41

**Comment by kul3r4** [comment]
I signed it!

2016-03-09 15:27 GMT+01:00 googlebot <notifications@github.com>:

> Thanks for your pull request. It looks like this may be your first
> contribution to a Google open source project. Before we can look at your
> pull request, you'll need to sign a Contributor License Agreement (CLA).
>
> [image: :memo:] *Please visit https://cla.developers.google.com/
> <https://cla.developers.google.com/> to sign.*
>
> Once you've signed, please reply here (e.g. I signed it!) and we'll
> verify. Thanks.
> ------------------------------
>
>    - If you've already signed a CLA, it's possible we don't have your
>    GitHub username or you're using a different email address. Check your
>    existing CLA data <https://cla.developers.google.com/clas> and verify
>    that your email is set on your git commits
>    <https://help.github.com/articles/setting-your-email-in-git/>.
>    - If you signed the CLA as a corporation, please let us know the
>    company's name.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ampproject/amp-by-example/pull/69#issuecomment-194318741>
> .
>



-- 
Chiara Chiappini

File: src/30_Advanced/all-videos-supported-by-amp.html [hunk 1]
```diff
@@ -0,0 +1,115 @@
+<!--
+  #### Introduction
+
+  A showcase off all the video types supported by AMP
```
Comment (sebastianbenz) [inline src/30_Advanced/all-videos-supported-by-amp.html]: Suggestion: "There many different ways to integrate videos in AMP HTML files. Here is a list of all currently supported video formats and platforms."

File: src/30_Advanced/all-videos-supported-by-amp.html [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+  <script async custom-element="amp-facebook" src="https://cdn.ampproject.org/v0/amp-facebook-0.1.js"></script>
+  <script async custom-element="amp-brightcove" src="https://cdn.ampproject.org/v0/amp-brightcove-0.1.js"></script>  
+  <meta name="viewport" content="width=device-width,minimum-scale=1,initial-scale=1">
+  <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
+  <script async src="https://cdn.ampproject.org/v0.js"></script>
+</head>
+<body>
+
+  <!-- #### Basic Usage -->
```
Comment (sebastianbenz) [inline src/30_Advanced/all-videos-supported-by-amp.html]: remove "Basic Usage"

Could you add a header to each sample? e.g.: `  <!-- #### Vimeo -->`

This has the positiv side effect that there is space between each video sample.

File: src/30_Advanced/all-videos-supported-by-amp.html [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    Find more examples [here](https://ampbyexample.com/components/amp-instagram/)
+  -->
+  <amp-instagram
+      data-shortcode="BCtYW7mhRV8"
+      width="1"
+      height="1"
+      layout="responsive">
+  </amp-instagram>
+  <!--
+    Use `amp-vine` for vine embed.
```
Comment (sebastianbenz) [inline src/30_Advanced/all-videos-supported-by-amp.html]: add s to embed.

**Comment by sebastianbenz** [comment]
Fantastic Chiara! Could you also add a sample for embedding a non-natively supported video platform via  amp-iframe? "You can always use `amp-iframe` to embed video from platforms without dedicated AMP components."

//cc @adewale

**Comment by sebastianbenz** [comment]
Two more things: 

1. Can we find a better title, e.g. 'Integrating Videos in AMP: an Overview'
1. The title is formatted incorrectly. You don't need to add dashs to the name, this happens automatically. Make sure to use quotes when creating a new sample, e.g. `gulp create --name 'Integrating Videos in AMP: an Overview' --dest src/30_Advanced`.

**Comment by kul3r4** [comment]
PTAL

**Comment by adewale** [comment]
I updated #41 to reference videos without dedicated AMP components which are loaded inside an amp-iframe.

**Comment by cramforce** [comment]
Nice!
Just a heads up that there are currently like 3 more of them in active pull requests.

File: src/30_Advanced/Integrating_Videos_in_AMP%3A_an_Overview.html [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+    Use `amp-brightcove` for brightcove videos.
+    Find more examples [here](https://ampbyexample.com/components/amp-brightcove/)
+  -->
+  <amp-brightcove
+                  data-account="906043040001"
+                  data-video-id="1401169490001"
+                  data-player-id="180a5658-8be8-4f33-8eba-d562ab41b40c"
+                  layout="responsive" width="480" height="270">
+  </amp-brightcove>
+  <!-- #### Iframe -->
```
Comment (sebastianbenz) [inline src/30_Advanced/Integrating_Videos_in_AMP%3A_an_Overview.html]: rename to amp-iframe

File: src/30_Advanced/Integrating_Videos_in_AMP%3A_an_Overview.html [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+  <script async custom-element="amp-brightcove" src="https://cdn.ampproject.org/v0/amp-brightcove-0.1.js"></script>  
+  <script async custom-element="amp-iframe" src="https://cdn.ampproject.org/v0/amp-iframe-0.1.js"></script>
+  <meta name="viewport" content="width=device-width,minimum-scale=1,initial-scale=1">
+  <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
+  <script async src="https://cdn.ampproject.org/v0.js"></script>
+</head>
+<body>
+
+  <!-- #### Video -->
```
Comment (sebastianbenz) [inline src/30_Advanced/Integrating_Videos_in_AMP%3A_an_Overview.html]: rename amp-video

File: src/30_Advanced/Integrating_Videos_in_AMP%3A_an_Overview.html [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+    <div fallback>
+      <p>Your browser doesn't support HTML5 video.</p>
+    </div>
+    <source type="video/mp4" src="video/tokyo.mp4">
+    <source type="video/webm" src="video/tokyo.webm">
+  </amp-video>
+  <!-- #### Instagram -->
+  <!--
+    Use `amp-instagram` for videos on Instagram.
+    Find more examples [here](https://ampbyexample.com/components/amp-instagram/)
```
Comment (sebastianbenz) [inline src/30_Advanced/Integrating_Videos_in_AMP%3A_an_Overview.html]: add a '.' after Find more examples here. The same applies to the other samples.

File: src/30_Advanced/Integrating_Videos_in_AMP%3A_an_Overview.html [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+      width="1"
+      height="1"
+      layout="responsive">
+  </amp-instagram>
+  <!-- #### Vine -->
+  <!--
+    Use `amp-vine` for vine embed.
+    Find more examples [here](https://ampbyexample.com/components/amp-vine/)
+  -->
+  <amp-vine width="400" height="400"
```
Comment (sebastianbenz) [inline src/30_Advanced/Integrating_Videos_in_AMP%3A_an_Overview.html]: can you make the vine sample responsive as well?

File: src/30_Advanced/Integrating_Videos_in_AMP%3A_an_Overview.html [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+                  layout="responsive" width="480" height="270">
+  </amp-brightcove>
+  <!-- #### Iframe -->
+  <!-- 
+    Use `amp-iframe` for native videos not supported in Amp.
+    Find more examples [here](https://ampbyexample.com/components/amp-iframe/)
+  -->
+  <amp-iframe width="500" height="281"
+                          layout="responsive"
+                          sandbox="allow-scripts allow-same-origin allow-popups allow-popups-to-escape-sandbox"
```
Comment (sebastianbenz) [inline src/30_Advanced/Integrating_Videos_in_AMP%3A_an_Overview.html]: change the sandbox args to: 'sandbox="allow-scripts allow-same-origin allow-popups' and add the allowfullscreen attribute (see my PR for amp-iframe).

Comment (sebastianbenz) [inline src/30_Advanced/Integrating_Videos_in_AMP%3A_an_Overview.html]: Even better: can you integrate an video player that is currently not supported by AMP, e.g. [ooyala]( http://support.ooyala.com/developers/documentation/concepts/player_v3_dev_amp.html) (thanks to @andreban for the link)

**Comment by kul3r4** [comment]
PTAL.

File: src/30_Advanced/Integrating_Videos_in_AMP%3A_an_Overview.html [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+  -->
+  <amp-brightcove
+                  data-account="906043040001"
+                  data-video-id="1401169490001"
+                  data-player-id="180a5658-8be8-4f33-8eba-d562ab41b40c"
+                  layout="responsive" width="480" height="270">
+  </amp-brightcove>
+  <!-- #### amp-iframe -->
+  <!-- 
+    Use `amp-iframe` for native videos not supported in Amp.
```
Comment (sebastianbenz) [inline src/30_Advanced/Integrating_Videos_in_AMP%3A_an_Overview.html]: nit: s/Amp/AMP/

File: src/30_Advanced/Integrating_Videos_in_AMP%3A_an_Overview.html [hunk 10]
```diff
@@ -0,0 +1,137 @@
...
+  -->
+  <amp-instagram
+      data-shortcode="BCtYW7mhRV8"
+      width="1"
+      height="1"
+      layout="responsive">
+  </amp-instagram>
+  <!-- #### amp-vine -->
+  <!--
+    Use `amp-vine` for vine embed.
```
Comment (sebastianbenz) [inline src/30_Advanced/Integrating_Videos_in_AMP%3A_an_Overview.html]: s/vine embed/Vine embeds/

File: src/30_Advanced/Integrating_Videos_in_AMP%3A_an_Overview.html [hunk 11]
```diff
@@ -0,0 +1,137 @@
...
+    Use `amp-youtube` for youtube videos.
+    Find more examples [here](https://ampbyexample.com/components/amp-youtube/).
+  -->
+  <amp-youtube width="480" height="270"
+                           layout=responsive
+                           data-videoid="lBTCB7yLs8Y" >
+  </amp-youtube>
+  <!-- #### amp-vimeo -->
+  <!--
+    Use `amp-vimeo` for vimeo videos.
```
Comment (sebastianbenz) [inline src/30_Advanced/Integrating_Videos_in_AMP%3A_an_Overview.html]: s/vimeo/Vimeo/

File: src/30_Advanced/Integrating_Videos_in_AMP%3A_an_Overview.html [hunk 12]
```diff
@@ -0,0 +1,137 @@
...
+    Use `amp-vine` for vine embed.
+    Find more examples [here](https://ampbyexample.com/components/amp-vine/).
+  -->
+  <amp-vine width="400" height="400"
+  layout="responsive"
+    data-vineid="MdKjXez002d">
+  </amp-vine>
+  <!-- #### amp-youtube -->
+  <!--
+    Use `amp-youtube` for youtube videos.
```
Comment (sebastianbenz) [inline src/30_Advanced/Integrating_Videos_in_AMP%3A_an_Overview.html]: s/youtube/YouTube/

File: src/30_Advanced/Integrating_Videos_in_AMP%3A_an_Overview.html [hunk 13]
```diff
@@ -0,0 +1,137 @@
...
+  -->
+  <amp-dailymotion
+    data-videoid="x2m8jpp"
+    layout="responsive"
+    data-ui-highlight="FF4081"
+    width="480" height="270">
+  </amp-dailymotion>
+  <!-- #### amp-twitter -->
+  <!-- 
+    Use `amp-twitter` for twitter videos.
```
Comment (sebastianbenz) [inline src/30_Advanced/Integrating_Videos_in_AMP%3A_an_Overview.html]: s/twitter/Twitter/

File: src/30_Advanced/Integrating_Videos_in_AMP%3A_an_Overview.html [hunk 14]
```diff
@@ -0,0 +1,137 @@
...
+    Find more examples [here](https://ampbyexample.com/components/amp-twitter/).
+  -->
+  <amp-twitter width="375" height="472"
+    layout="responsive"
+    data-tweetid="707569406105092096">
+  </amp-twitter>
+  <!-- #### amp-facebook -->
+  <!-- 
+    Use `amp-facebook` for facebook videos.
```
Comment (sebastianbenz) [inline src/30_Advanced/Integrating_Videos_in_AMP%3A_an_Overview.html]: s/facebook/Facebook/

File: src/30_Advanced/Integrating_Videos_in_AMP%3A_an_Overview.html [hunk 15]
```diff
@@ -0,0 +1,137 @@
...
+    Find more examples [here](https://ampbyexample.com/components/amp-facebook/).
+  -->
+  <amp-facebook width="552" height="310"
+                    layout="responsive"
+                    data-href="https://www.facebook.com/zuck/videos/10102509264909801/">
+  </amp-facebook>
+  <!-- #### amp-brightcove -->
+  <!-- 
+    Use `amp-brightcove` for brightcove videos.
```
Comment (sebastianbenz) [inline src/30_Advanced/Integrating_Videos_in_AMP%3A_an_Overview.html]: s/brightcove/Brightcove/

**Comment by sebastianbenz** [comment]
We should use the correct upper and lowercase spelling of the video platforms (some companies are picky about this). 

Question: does it make sense to sort the video platform alphabetically with amp-video at the beginning and amp-iframe as the last element?

File: src/30_Advanced/Integrating_Videos_in_AMP%3A_an_Overview.html [hunk 16]
```diff
@@ -0,0 +1,137 @@
...
+  -->
+  <amp-instagram
+      data-shortcode="BCtYW7mhRV8"
+      width="1"
+      height="1"
+      layout="responsive">
+  </amp-instagram>
+  <!-- #### amp-vine -->
+  <!--
+    Use `amp-vine` for Vine Embeds.
```
Comment (sebastianbenz) [inline src/30_Advanced/Integrating_Videos_in_AMP%3A_an_Overview.html]: s/Embeds/embeds/

File: src/30_Advanced/Integrating_Videos_in_AMP%3A_an_Overview.html [hunk 17]
```diff
@@ -0,0 +1,137 @@
...
+    Find more examples [here](https://ampbyexample.com/components/amp-vimeo/).
+  -->
+  <amp-vimeo
+      data-videoid="27246366"
+      layout="responsive"
+      width="16" height="9">
+  </amp-vimeo>
+  <!-- #### amp-dailymotion -->
+  <!--
+    Use `amp-dailymotion` for Dailymotion videos.
```
Comment (sebastianbenz) [inline src/30_Advanced/Integrating_Videos_in_AMP%3A_an_Overview.html]: s/Dailymotion/dailymotion/ they don't write uppercase at the beginning

**Comment by kul3r4** [comment]
If we want to order them alphabetically, then I would say amp-youtube comes first. Otherwise I don't understand the ordering you are suggesting and it's just confusing if a part of it is ordered and some not.

**Comment by sebastianbenz** [comment]
My thinking was: 

- amp-video first as it is the amp-native way
- amp-iframe last as it is your last chance ;-)

But you're right - let's stay with simple alphabetical ordering.

**Comment by kul3r4** [comment]
Alphabetical a-z order. PTAL.

**Comment by sebastianbenz** [comment]
Yay! Little nit: amp-twitter < amp-video

**Comment by kul3r4** [comment]
ops! fixed.

**Comment by kul3r4** [comment]
ops! fixed.

**Comment by sebastianbenz** [comment]
LGTM

**Pull request merged**